### PR TITLE
[13.0][FIX] project_stock: Update the onchange of the task so that it does not behave incorrectly when creating stock moves.

### PR DESCRIPTION
**Pull request by victoralmau**
Update the onchange of the task so that it does not behave incorrectly when creating stock moves.

Please @pedrobaeza can you review it?

@Tecnativa TT35602

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This if fixing the consequences, but not the real root cause. Why some of them are not filled? Are you doing things through onchanges and the fields don't exist? Maybe you need to give read permission to some master data like locations, and only hide elements, not totally remove them (group inside the arch, not in the view).

**Comment by victoralmau** [comment]
> This if fixing the consequences, but not the real root cause. Why some of them are not filled? Are you doing things through onchanges and the fields don't exist? Maybe you need to give read permission to some master data like locations, and only hide elements, not totally remove them (group inside the arch, not in the view).

Ok sorry, you're right. The correct change is already applied.

File: project_stock/models/project_task.py [hunk 1]
```diff
@@ -101,17 +101,19 @@ def _compute_unreserve_visible(self):
 
     @api.onchange("picking_type_id")
     def onchange_picking_type(self):
```
Comment (pedrobaeza) [inline project_stock/models/project_task.py]: Naming guidelines:

```suggestion
    def _onchange_picking_type_id(self):
```

File: project_stock/models/project_task.py [hunk 2]
```diff
@@ -101,17 +101,19 @@ def _compute_unreserve_visible(self):
...
-        self.location_id = self.picking_type_id.default_location_src_id.id
-        self.location_dest_id = self.picking_type_id.default_location_dest_id.id
+        if self.picking_type_id:
+            self.location_id = self.picking_type_id.default_location_src_id.id
+            self.location_dest_id = self.picking_type_id.default_location_dest_id.id
 
-    @api.onchange("location_id", "move_ids")
+    @api.onchange("location_id")
     def _onchange_location(self):
```
Comment (pedrobaeza) [inline project_stock/models/project_task.py]:
```suggestion
    def _onchange_location_id(self):
```

**Review by pedrobaeza — APPROVED** [review]

**Comment by victoralmau** [comment]
Included in https://github.com/OCA/project/pull/924

**Pull request closed without merging**